src/App.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react';
-import './App.css';
+import React, {useEffect} from 'react';
+import {
+  RouterProvider,
+} from "react-router-dom";
+import {useAppDispatch} from "./Hooks/hook";
+import auth from "./Firebase";
+import {userAuth, userLogOut, userToken} from "./Redux/Store/auth/authSlice";
+import axios from "axios";
+import {appRouter} from "./Routes/Router";
+
+
+function App():JSX.Element {
+  const dispatch = useAppDispatch()
+  useEffect(()=> {
+    auth.onAuthStateChanged(authUser => {
+      if(authUser){
+        // console.log(authUser);
+        authUser.getIdTokenResult().then(result=> {
+          // console.log(result);
+          const token:string = result.token
+          dispatch(userToken(token))
+          const request: Promise<void> = axios.get("http://localhost:3000/userAuth/verifyUser",{
```

Comment:
harcoded url shouldn't exists here.

Reply:
Done! now I have an Axios instance with my base url and authentication token, so I don't need to provide it every time I make a request.